File: day3/rounds.py
#!/opt/homebrew/bin/python3

# Build P(riority) = { 'a': 1, ..., 'z': 25, 'A': 26, ..., 'Z': 52 }
P = [{ chr(x):x-ord(A)+N for x in range(ord(A), ord(A) + 26) } for A,N in { 'a':1, 'A':27 }.items()]
P = P[0] | P[1]

total1, total2, group = 0, 0, []

file = open("input.txt", "r")
for line in file:
  l = len(line)
  total1 += P[( set(line[:l//2]) & set(line[l//2:].strip()) ).pop()]

  group.append(line.strip())
  if len(group) == 3:
    total2 += P[( set(group[0]) & set(group[1]) & set(group[2]) ).pop()]
    group = []

file.close()
print(f"total round 1: {total1}")
print(f"total round 2: {total2}")

File: day2/rounds.pl
#!/opt/homebrew/bin/perl

use strict;

my $round = 1; # exercice round : 1 or 2

my $total = 0;
my %play_index = ('A', 0, 'B', 1, 'C', 2, 'X', 0, 'Y', 1, 'Z', 2);
my @plays = ( [4, 8, 3], [1, 5, 9], [7, 2, 6] );

@plays = map { [ sort { $a <=> $b } @{$_} ] } @plays if ($round == 2);

open(INPUT, 'input.txt') or die "file error";
while ( my @play = split(' ', <INPUT>) )  {
    $total += $plays[$play_index{$play[0]}][$play_index{$play[1]}];
}
close(INPUT);

print "$total\n";
